component/otelcol/processor/memorylimiter/memorylimiter.go

```diff
@@ -0,0 +1,120 @@
+// Package memorylimiter provides an otelcol.processor.memory_limiter component.
+package memorylimiter
```

Comment:
Don't forget to include this in github.com/grafana/agent/component/all so it gets defined :)